File: 0386-lexicographical-numbers/0386-lexicographical-numbers.cpp
class Solution {
public:
    vector<int> lexicalOrder(int n) {
        vector<int> res;
        int curr = 1;
        while(res.size() < n) {
            res.push_back(curr);
            if(curr*10 <= n) curr *= 10;
            else {
                while(curr == n || curr % 10 == 9) {
                    curr /= 10;
                }
                curr += 1;
            }
        }

        return res;
    }
};

File: 0386-lexicographical-numbers/0386-lexicographical-numbers.py
class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        return sorted(list(range(1, n+1)), key=lambda x:str(x))

with open("user.out", "w") as f:
    inputs = map(loads, stdin)
    for nums in inputs:
        print(str(Solution().lexicalOrder(nums)).replace(" ", ""), file=f)
exit(0)